### ISSUE TITLE
Metadaten View auf Arbeitsbereichsebene

### ISSUE THREAD

**Issue by jurei733** [iqb-berlin/studio-lite#677]
Für einen Arbeitsbereich kann eine Metadatenübersicht erstellt werden mit Filter Möglichkeiten

**Comment by Annika-Schlotz** [comment]
Einstellungen -> Berichte -> Metadaten zeigt den Bericht
@jurei733 reicht das so nicht aus?

**Comment by jurei733** [comment]
Nein, der vorhandene Metadatenbericht reicht noch nicht aus, um #677 vollständig abzudecken.

Die Übersicht und der Excel-Download sind bereits vorhanden. Die aktuelle Filterfunktion ist jedoch nur eine allgemeine Freitextsuche über die gesamte Tabelle. Mit den im Issue genannten Filtermöglichkeiten waren gezielte Filter für einzelne Metadatenspalten gemeint, die miteinander kombiniert werden können – beispielsweise:

- Fach = Mathematik oder Deutsch
- Jahrgangsstufe = 8
- Aufgabentyp = Multiple Choice

Die zuvor ausgewählten Aufgaben bilden dabei die Grundmenge des Berichts. Die fachlichen Spaltenfilter schränken diese Auswahl anschließend weiter ein. Die bestehende allgemeine Tabellensuche kann zusätzlich erhalten bleiben.

## Vorgeschlagene Filterarten

Die angebotenen Bedingungen sollen sich nach dem Datentyp der jeweiligen Metadatenspalte richten:

- Textfelder: „enthält“, „enthält nicht“, „ist gleich“ und „ist nicht gleich“
- Felder mit vorgegebenen Werten: Auswahl eines oder mehrerer Werte
- Zahlenfelder: „gleich“, „größer als“, „kleiner als“ und Wertebereich
- Datumsfelder: „vor“, „nach“ und Zeitraum
- Ja/Nein-Felder: „Ja“ oder „Nein“
- Für alle Felder: „ist leer“ und „ist nicht leer“

Über „Filter hinzufügen“ werden nacheinander die Metadatenspalte, die passende Bedingung und der gewünschte Wert ausgewählt. Nach dem Anwenden erscheint der Filter als sichtbarer Chip. Der Filter kann über den Chip bearbeitet oder einzeln entfernt werden. Zusätzlich sollen alle aktiven Filter gemeinsam zurückgesetzt werden können.

## Verknüpfung mehrerer Filter

Mehrere ausgewählte Werte innerhalb derselben Spalte werden mit ODER verknüpft. Der Filter

> Fach = Mathematik oder Deutsch

findet daher Datensätze, bei denen mindestens einer der beiden Werte hinterlegt ist.

Filter aus unterschiedlichen Spalten werden mit UND verknüpft. Bei

> Fach = Mathematik oder Deutsch
> UND Jahrgangsstufe = 8
> UND Aufgabentyp = Multiple Choice

werden nur Datensätze angezeigt, die alle drei Bedingungen erfüllen.

Wertebereiche wie „Jahrgangsstufe von 5 bis 8“ werden innerhalb eines gemeinsamen Bereichsfilters abgebildet.

## Aufgaben- und Item-Metadaten

Die Spaltenfilter sollen sowohl in der Ansicht „Aufgaben“ als auch in der Ansicht „Items“ verfügbar sein.

In der Aufgabenansicht entspricht jede Zeile einer Aufgabe. Neben der Aufgabenkennung werden die im Metadatenprofil des Arbeitsbereichs definierten Aufgaben-Metadaten als Spalten dargestellt.

In der Item-Ansicht entspricht jede Zeile einem Item. Dort werden beispielsweise Aufgabe, Item-ID, Variable und Notiz sowie die konfigurierten Item-Metadaten angezeigt.

Beide Ansichten besitzen eigene Filter. Beim Wechsel zwischen „Aufgaben“ und „Items“ können die gesetzten Filter der jeweiligen Ansicht erhalten bleiben.

Die in den Mockups dargestellten fachlichen Spalten wie „Kompetenzbereich“ oder „Schwierigkeitsgrad“ sind nur Beispiele. Tatsächlich angezeigte Spalten ergeben sich dynamisch aus den für den Arbeitsbereich konfigurierten Metadatenprofilen.

## Sortierung, Ergebnisanzeige und Download

Die Spalten bleiben auf- und absteigend sortierbar. Oberhalb der Tabelle wird angezeigt, wie viele Datensätze nach Anwendung der Filter übrig bleiben, beispielsweise:

> 9 von 186 Aufgaben

Der primäre Excel-Download soll die aktuell gefilterten Ergebnisse enthalten. Dadurch stimmen Tabellenansicht und exportierte Daten überein.

Falls zusätzlich ein vollständiger Export benötigt wird, sollte dafür eine zweite, eindeutig bezeichnete Aktion angeboten werden:

- „Gefilterte Ergebnisse herunterladen“
- „Alle ausgewählten Aufgaben herunterladen“

In der Item-Ansicht wären entsprechend die gefilterten Items beziehungsweise alle Items der ausgewählten Aufgaben zu exportieren.

## Konzeptionelle Mockups

### 1. Ausgangszustand

Alle zuvor ausgewählten Aufgaben werden angezeigt. Es sind noch keine fachlichen Spaltenfilter aktiv.

<img width="1199" height="675" alt="Mockup: Ausgangszustand ohne Filter" src="https://github.com/user-attachments/assets/20e32714-44d5-4b66-9e10-8c8093639697" />

### 2. Filter hinzufügen

Nach dem Anklicken von „Filter hinzufügen“ werden Metadatenspalte, Bedingung und Wert ausgewählt. Im Beispiel werden innerhalb der Spalte „Fach“ die Werte „Mathematik“ und „Deutsch“ mit ODER verknüpft.

<img width="1199" height="675" alt="Mockup: Filter hinzufügen" src="https://github.com/user-attachments/assets/8957e0b2-ab99-434d-aab8-4f9550b9de9f" />

### 3. Kombinierte Aufgabenfilter

Die aktiven Filter werden als Chips dargestellt. Filter unterschiedlicher Spalten sind UND-verknüpft. Die Ergebnisanzahl und der Download beziehen sich auf die gefilterte Tabelle.

<img width="1199" height="675" alt="Mockup: kombinierte Aufgabenfilter" src="https://github.com/user-attachments/assets/94aa4382-fe6e-4443-89be-f86446a6c4ea" />

### 4. Separate Item-Ansicht

Die Item-Ansicht besitzt eigene Spalten und Filter. Jede Tabellenzeile entspricht einem Item innerhalb der zuvor ausgewählten Aufgaben.

<img width="1199" height="675" alt="Mockup: separate Item-Ansicht" src="https://github.com/user-attachments/assets/098a8aa9-523a-4655-8da7-608f1a09c763" />

Die Mockups dienen als Diskussionsgrundlage. Die konkrete Gestaltung kann bei der Umsetzung an die bestehenden Studio-Komponenten angepasst werden.

Die zentrale Filteranforderung aus #677 ist weiterhin offen. Das Issue sollte daher nicht geschlossen werden.
